Super tests - mama, look, I am coding like a pro

helgamuller opened the pull request:
1. Added all missed checks
2. Added test for DELETE endpoint
3. Added data clean up(user deleting) where it necessary
4. Improved framework up to Senior level

nobugs-me commented on src/test/java/iteration1/CreateAccountTest.java:
```diff
@@ -1,81 +1,42 @@
+                        .anyMatch(acc -> acc.getId() == accountId));
 
+        int userId = AdminSteps.getUserId(userRequest);
+        AdminSteps.deleteUser(userId);
```
Удаление нужно вынести за пределы теста, например в AfterEach

helgamuller replied:
Done

nobugs-me reviewed (changes requested):
Привет! 

Все отлично, вижу, что структура освоена и применена как ожидалось. 

Единственное, нужно вынести очищение окружения за пределы тестов.

helgamuller commented:
- Moved user removal into @AfterEach
 - Changed badRequest spec due to changes in server response when you try to create a user with invalid creds. It returned raw messge but has been changed and now returns an array
